smartsim/_core/_cli/build.py

```diff
@@ -125,212 +112,72 @@ def build_database(
-    }
-    missing, conflicts = _assess_python_env(
-        torch_deps,
-        package_pinning="==",
-        validate_installed_version=_create_torch_version_validator(
-            with_suffix=device_suffix
-        ),
-    )
 
-    if len(missing) == len(torch_deps) and not conflicts:
-        # All PyTorch deps are not installed and there are no conflicting
-        # python packages. We can try to install torch deps into the current env.
-        logger.info(
-            "Torch version not found in python environment. "
-            "Attempting to install via `pip`"
-        )
-        pip(
-            "install",
-            "--extra-index-url",
-            f"https://download.pytorch.org/whl/{device.torch_suffix()}",
-            *(f"{package}=={version}" for package, version in torch_deps.items()),
-        )
-    elif missing or conflicts:
-        logger.warning(_format_incompatible_python_env_message(missing, conflicts))
-
-
-def _create_torch_version_validator(
-    with_suffix: str,
-) -> t.Callable[[str, t.Optional[Version_]], bool]:
-    def check_torch_version(package: str, version: t.Optional[Version_]) -> bool:
-        if not BuildEnv.check_installed(package, version):
-            return False
-        # Default check only looks at major/minor version numbers,
-        # Torch requires we look at the patch as well
-        installed = BuildEnv.get_py_package_version(package)
-        if with_suffix and with_suffix not in installed.patch:
-            raise VersionConflictError(
-                package,
-                installed,
-                version or Version_(f"X.X.X{with_suffix}"),
-                msg=(
-                    f"{package}=={installed} does not satisfy device "
-                    f"suffix requirement: {with_suffix}"
-                ),
-            )
-        return True
-
-    return check_torch_version
-
-
-def _check_packages_in_python_env(
-    packages: t.Mapping[str, t.Optional[Version_]],
-    package_pinning: _TPinningStr = "==",
-    validate_installed_version: t.Optional[
-        t.Callable[[str, t.Optional[Version_]], bool]
-    ] = None,
+def build_redis_ai(
+    platform: Platform,
+    mlpackages: MLPackageCollection,
+    build_env: BuildEnv,
+    verbose: bool,
 ) -> None:
-    # TODO: Do not like how the default validation function will always look for
-    #       a `==` pinning. Maybe turn `BuildEnv.check_installed` into a factory
-    #       that takes a pinning and returns an appropriate validation fn?
-    validate_installed_version = validate_installed_version or BuildEnv.check_installed
-    missing, conflicts = _assess_python_env(
-        packages,
-        package_pinning,
-        validate_installed_version,
-    )
+    logger.info("Building RedisAI and backends...")
+    RAIBuilder = RedisAIBuilder(platform, mlpackages, build_env, verbose)
+    RAIBuilder.build()
+    RAIBuilder.cleanup_build()
```

Comment:
I'm a little bit wary of doing either of those two things, because if there was an Exception thrown during the build, we need to have the build directory around to debug the problem. I suppose we could go the context manager route and only cleanup on success. What's your preffered path?

Reply:
Could also just make it the responsibility of the builder to cleanup the build directory for everything